### PR TITLE
METRON-2020: Running run_end_to_end.sh with docker give warning if bash < 4.0

### PR DESCRIPTION
**Pull request by JonZeolla**
## Contributor Comments
Added a check to the `analyze_results.sh` and `run_end_to_end.sh` scripts to verify the scripts are being run with bash >= 4.  Technically this is only needed due to the associative arrays in `analyze_results.sh` but I pushed it to `run_end_to_end.sh` as well to provide faster feedback since we know the analyzation at the end will fail if they're running bash 3 or earlier.

### Testing
- Run with bash < 4 (`/bin/bash` on a mac) and expect failure.
- Run with bash > 4 and expect success.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron's Bro kafka writer plugin.

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [X] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [X] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [X] Have you included steps or a guide to how the change may be verified and tested manually?
- [X] Have you ensured that the full suite of tests and checks have been executed via:
  ```
  bro-pkg test $GITHUB_USERNAME/metron-bro-plugin-kafka --version $BRANCH
  ```
- [ ] Have you written or updated unit tests and or integration tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] Have you verified the basic functionality of the build by building and running locally with Apache Metron's [Vagrant full-dev environment](https://github.com/apache/metron/tree/master/metron-deployment/development/centos6) or the equivalent?

### CODE REVIEW

**Comment by JonZeolla** [comment]
Ok this is stable now.  Confirmed working in `run_end_to_end.sh` and `analyze_results.sh` separately.

File: docker/run_end_to_end.sh [hunk 1]
```diff
@@ -33,6 +33,12 @@ function help {
   echo " "
 }
```
Comment (ottobackwards) [inline docker/run_end_to_end.sh]: I think the -help message should include the bash requirement

Comment (JonZeolla) [reply docker/run_end_to_end.sh]: How's this?

Comment (ottobackwards) [reply docker/run_end_to_end.sh]: Can we call help when we fail for the wrong version?

**Comment by ottobackwards** [comment]
+1

**Pull request closed without merging**